### ISSUE TITLE
`unnecessary_nesting_linter` and `brace_linter` are contradictory

### ISSUE THREAD

**Issue by TanguyBarthelemy** [r-lib/lintr#2823]
I can't set `unnecessary_nesting_linter` and `brace_linter` simultaneously in the following code. This is just an example:

```r
nbr <- 1:10

lapply(
    X = nbr,
    FUN = \(k) { #Reduce the nesting of this statement by removing the braces {}.
        paste(
            "2 x",
            k,
            "=",
            2 * k
        )
    }
)

lapply(
    X = nbr,
    FUN = \(k) paste( #Use curly braces for any function spanning multiple lines.
        "2 x",
        k,
        "=",
        2 * k
    )
)
```

What is the best way to deal with the style rules? Should you remove or add braces, or rewrite the code?

**Comment by TanguyBarthelemy** [comment]
I'm using {lintr} in version 3.2.0 and R 4.4.2.

**Comment by TanguyBarthelemy** [comment]
Ok I didn't know that `function()` was preferred to `\()`.
Thank you for your prompt response!

**Comment by MichaelChirico** [comment]
see https://style.tidyverse.org/functions.html#anonymous-functions

**Comment by TanguyBarthelemy** [comment]
Yes, this has been corrected. But the message doesn't indicate the use of function() instead of \(). Perhaps add a message to that effect? But I don't know in which linter

**Comment by MichaelChirico** [comment]
That won't come from here, follow #2667 for when that rule would be implemented (it will be a default linter)
